src/main/common/utils/numericUtils.ts

```diff
@@ -14,3 +14,37 @@ export function toNumberOrUndefined (value: any): number {
 
   return isNaN(numberVal) ? undefined : numberVal
 }
+
+export function containsAValidComma (input: string): boolean {
+  const output: string[] = input.split(',')
+
+  return output[output.length - 1].length >= 3
+}
+
+export function toAValidFloatOrOriginalValue (input: string): any {
+  const output: string = this.containsAValidComma(input) ? input.replace(new RegExp(/,/g), '') : input
+
+  const numberValue = parseFloat(output)
+
+  if (!output.includes(',') && !isNaN(numberValue) && !this.isExponential(numberValue)) {
+    return numberValue
+  } else {
+    return output
+  }
+}
+
+export function isExponential (numberValue: number) {
```

Comment:
Do not export that function unless needed. If you exposing it make it well tested.

Reply:
Also why do we need to handle scientific notation? I would just discard it as invalid value. No sane person would use scientific notation when claiming for money.